### ISSUE TITLE
dict.iteritems not spported in python 3

### ISSUE THREAD

**Issue by matejv** [ArnesSI/pynetdot#12]
It seems there are still some bugs when running under python 3.

Here is one:

```Python traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)

~/temp/devel/netdot_extra/netdot_device.py in save_attribute(self, attribute_name, value, overwrite)
    165             da.device = self.device
    166         da.value = value
--> 167         da.save()
    168         return da
    169 

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/netdot.py in save(self)
    192         id is not set.
    193         """
--> 194         if not self.is_dirty():
    195             # There were no changes
    196             return True

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/netdot.py in is_dirty(self)
    156         if not self.id:
    157             return True
--> 158         return bool(self.get_dirty_fields())
    159 
    160     def display(self, view='all'):

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/netdot.py in get_dirty_fields(self)
    147         changed_fields = dict([
    148             (key, value)
--> 149             for key, value in self._original_state.iteritems()
    150             if value != new_state[key]
    151         ])

AttributeError: 'dict' object has no attribute 'iteritems'
```

**Comment by matejv** [comment]
Possibly another:

```
In [39]: for ar in arnes.records:
    ...:     if ar.devices:
    ...:         print(ar)
    ...:         

/home/matej/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/serializer.py:27: UserWarning: One or more invalid characters found in XML string received from netdot API. They were removed. Unpredictable side effects possible.
  warnings.warn('One or more invalid characters found in XML string received from netdot API. They were removed. Unpredictable side effects possible.')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-39-3f41597dca8c> in <module>()
      1 for ar in arnes.records:
----> 2     if ar.devices:
      3         print(ar)
      4 

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/models/base.py in devices(self)
   2442     def devices(self):
   2443         cls = getattr(pynetdot.models, 'Device')
-> 2444         return cls.search(name=self.id)
   2445 
   2446     @property

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/netdot.py in search(cls, **kwargs)
     72         if not kwargs:
     73             raise Exception('Need to specify search parameters')
---> 74         return cls._search(**kwargs)
     75 
     76     @classmethod

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/netdot.py in _search(cls, **kwargs)
     82         if not response.ok:
     83             response.raise_for_status()
---> 84         xml = parse_xml(response.content)
     85         results = []
     86         for c in xml:

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/serializer.py in parse_xml(data)
     11         # on python3 response will be bytes() and needs to be decoded as utf-8
     12         data = data.decode('utf-8')
---> 13     data = _xml_pre_clean(data)
     14     root = ET.fromstring(data)
     15     return root

~/.virtualenvs/pynetdot/lib/python3.6/site-packages/pynetdot/serializer.py in _xml_pre_clean(data)
     26     if any(c in data for c in INVALID_CHARS):
     27         warnings.warn('One or more invalid characters found in XML string received from netdot API. They were removed. Unpredictable side effects possible.')
---> 28         return data.translate(None, INVALID_CHARS)
     29     else:
     30         return data

TypeError: translate() takes exactly one argument (2 given)
```

**Comment by matejv** [comment]
this last one was fixed by #13
